### PR TITLE
Fix the wrong bus usage and remove multi mem-backend on s390x

### PR DESCRIPTION
**Pull request by fbq815**
virtio_fs: update bus to 'virtual-css' and remove mutl memory backend
on s390x

ID: 2382

Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
#https://github.com/avocado-framework/avocado-vt/pull/3911

Test result:
RHEL8:
JOB ID     : 3d3a9b9d0452a0e8d8b50d2bf26509cd12820640
JOB LOG    : /root/avocado/job-results/job-2024-05-15T04.45-3d3a9b9/job.log
 (1/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.s390-virtio: STARTED
 (1/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.s390-virtio: PASS (458.68 s)
 (2/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.multi_fs_sources.s390-virtio: STARTED
 (2/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.multi_fs_sources.s390-virtio:PASS (44.03 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-05-15T04.45-3d3a9b9/results.html
JOB TIME   : 504.40 s

RHEL9:
JOB ID     : 715b1f11b4edefe25a13f933d1e6227c83d7e768
JOB LOG    : /root/avocado/job-results/job-2024-05-15T05.10-715b1f1/job.log
 (1/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: STARTED
 (1/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.s390x.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads.s390-virtio: PASS (983.83 s)
 (2/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.s390-virtio: STARTED
 (2/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.s390-virtio: PASS (447.75 s)
 (3/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.multi_fs_sources.s390-virtio: STARTED
 (3/3) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.multi_fs_sources.s390-virtio: PASS (40.98 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-05-15T05.10-715b1f1/results.html
JOB TIME   : 1474.60 s

**Comment by fbq815** [comment]
@PaulYuuu please help me review this patch

File: qemu/tests/virtio_fs_hotplug.py [hunk 1]
```diff
@@ -240,7 +240,12 @@ def plug_fs_devices(action, plug_devices):
             if isinstance(dev, qdevices.CharDevice):
                 dev.set_param("server", "off")
             if isinstance(dev, qdevices.QDevice) and action == "hotplug":
-                parent_bus = 'pcie_extra_root_port_%s' % index if params['machine_type'] == 'q35' else 'pci.0'
+                if params['machine_type'] == 'q35':
```
Comment (PaulYuuu) [inline qemu/tests/virtio_fs_hotplug.py:243]: aarch64 also uses PCIE topology.

Comment (fbq815) [reply qemu/tests/virtio_fs_hotplug.py]: updated

Comment (fbq815) [reply qemu/tests/virtio_fs_hotplug.py]: @PaulYuuu please help me review this patch

**Comment by fbq815** [comment]
Test result:
JOB ID     : fd30f6edcec845d634ccdd7797af0ea60aab4e23
JOB LOG    : /root/avocado/job-results/job-2024-05-21T01.56-fd30f6e/job.log
 (1/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.s390-virtio: STARTED
 (1/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.one_fs_source.with_repetition.s390-virtio: PASS (456.91 s)
 (2/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.multi_fs_sources.s390-virtio: STARTED
 (2/2) Host_RHEL.m8.u10.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.10.0.s390x.io-github-autotest-qemu.virtio_fs_hotplug.plug.multi_fs_sources.s390-virtio: PASS (43.36 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2024-05-21T01.56-fd30f6e/results.html
JOB TIME   : 501.96 s

**Review by PaulYuuu — APPROVED** [review]

**Comment by fbq815** [comment]
@hellohellenmao @zhenyzha @MiriamDeng @xiagao please help me check if this work on your platform thx

**Comment by zhenyzha** [comment]
LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>
 
```
(1/5) virtio_fs_hotplug.plug.one_fs_source.arm64-pci: PASS (134.36 s)
 (2/5) virtio_fs_hotplug.plug.one_fs_source.with_repetition.arm64-pci: PASS (1436.69 s)
 (3/5) virtio_fs_hotplug.plug.multi_fs_sources.arm64-pci: PASS (94.82 s)
 (4/5) virtio_fs_hotplug.unplug.one_fs_source.arm64-pci: PASS (53.18 s)
 (5/5) virtio_fs_hotplug.unplug.multi_fs_sources.arm64-pci: PASS (67.55 s)
```

**Review by zhenyzha — APPROVED** [review]

**Comment by MiriamDeng** [comment]
The test is on going and will share the test result today.

**Comment by fbq815** [comment]
@PaulYuuu could you please help me merge this patch?

**Pull request merged**